src/converter/converter.py

```diff
@@ -99,7 +96,9 @@ def stage_order(self, element):
 
     def _compute_stage_order(self):
         stages = list(self.get_benchmark_stages().values())
-        stage_order_map = {self.get_stage_id(stage): pos for pos, stage in enumerate(stages)}
+        stage_order_map = {
+            self.get_stage_id(stage): pos for pos, stage in enumerate(stages)
+        }
         # FIXME very rudimentary computation of ordering
         # FIXME Might be more complex in future benchmarking scenarios
         # Assuming the order in which stages appear in the benchmark YAML is the actual order of the stages during execution
```

Comment:
I think we should print this as an INFO message and add that to the README, as it is a strong assumption we need to be aware of

Reply:
What about inferring the order properly from the `dag` instead? `nx.topological_sort(G)` should work?

Reply:
The code to infer the `stage_dag` from the benchmark model:

```
def get_output_stage(out_id: str, benchmark: model.Benchmark) -> model.Stage:
    """Get stage that returns output with out_id"""
    stage_by_output: dict = {}
    for stage in benchmark.stages:
        stage_by_output.update({out.id: stage for out in stage.outputs})
    return stage_by_output.get(out_id)

def build_stage_dag(benchmark: model.Benchmark) -> nx.DiGraph:
    g = nx.DiGraph()
    for stage in benchmark.stages:
        g.add_node(stage.id)
        input_ids = [input_id for input in stage.inputs for input_id in input.entries]
        dep_stages = [get_output_stage(input_id, benchmark) for input_id in input_ids]
        for dep in dep_stages:
            g.add_edge(dep.id, stage.id)
    return g
```
Then we can directly compute the stage order:
```
def compute_stage_order(stage_dag: nx.DiGraph) -> List:
  try:
     topological_order = list(nx.topological_sort(stage_dag))
  except nx.NetworkXUnfeasible:
      raise InputError("The stage graph has a cyclic dependencies. This benchmark can not be resolved")
  return topological_order
```
Happy to make a PR with these changes, once the 3 open PRs are merged

Reply:
Thanks a lot for the code. I refactored a bit the interface and included this logic of computing the stage order.